torch/csrc/jit/passes/quantization.cpp

```diff
@@ -226,7 +226,13 @@ Node* InsertObserversHelper::insertObserverFor(
         {});
     // Insert call to observer's forward
     Node* call = g->insertMethodCall("forward", forward_matched_schema)->node();
-    call->output()->setDebugName(v->debugName() + ".observed");
+    call->output()->copyMetadata(v);
+
+    // Replace v with the output of observer
+    v->replaceAllUsesWith(call->output());
```

Comment:
Do we require all observers to return unchanged input value? Is it how they are implemented now?

Reply:
Yes that is a contract we have